service/reducers/entities/files.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {FilesTypes, UsersTypes} from 'service/constants';
+
+function files(state = {}, action) {
+    switch (action.type) {
+    case FilesTypes.RECEIVED_FILES_FOR_POST: {
+        const filesById = action.data.reduce((filesMap, file) => {
+            return {...filesMap,
+                [file.id]: file
+            };
+        }, {});
+        return {...state,
+            ...filesById
+        };
+    }
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
+function fileIdsByPostId(state = {}, action) {
```

Comment:
Yeah, they'll use the `filenames` field on the first load, but as soon as you call the `getFileInfosForPost` api, that field is cleared and `file_ids` is populated.. I don't think we should bother including any handling for this in the RN app though. That field will only be set on posts made before 3.5 that haven't been viewed since then, so it'll be pretty rare